voice-assistant/m5stack-atom-echo.yaml

```diff
@@ -38,53 +35,53 @@ microphone:
   - platform: i2s_audio
     id: echo_microphone
     i2s_din_pin: GPIO23
+    adc_type: external
+    pdm: true
 
-media_player:
+speaker:
   - platform: i2s_audio
-    name: None
-    id: echo_audio
-    dac_type: external
+    id: echo_speaker
     i2s_dout_pin: GPIO22
+    dac_type: external
     mode: mono
 
 voice_assistant:
   microphone: echo_microphone
+  speaker: echo_speaker
   on_start:
     - light.turn_on:
         id: led
         blue: 100%
         red: 0%
         green: 0%
+        effect: none
   on_tts_start:
     - light.turn_on:
         id: led
         blue: 0%
         red: 0%
         green: 100%
+        effect: none
   on_tts_end:
     - light.turn_on:
         id: led
         blue: 0%
         red: 0%
         green: 100%
-    - media_player.play_media: !lambda return x;
```

Comment:
I have recently intalled Voice Assistant on M5 Echo and noticed that speaker is not playing Voice Assistant responses. I see that `media_player.play_media` accepts an URL which `x` - the argument of `on_tts_end` - is. But `speaker.play` accepts data bytes.

Is there a way to download file and convert it to bytes in esphome?

Reply:
Ahh, I see. From voice_assistant_version=2 it should use the speaker to play back the TTS response received via UDP connection (as per https://github.com/home-assistant/core/pull/92052). However by M5 Echo reports `voice_assistant_version=3`, but no response is being played through speakers.